napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1609 @@
+            applied to each shape otherwise the same value will be used for all
+            shapes.
+        edge_width : float | list
+            thickness of lines and edges. If a list is supplied it must be the
+            same length as the length of `data` and each element will be
+            applied to each shape otherwise the same value will be used for all
+            shapes.
+        edge_color : str | tuple | list
+            If string can be any color name recognized by vispy or hex value if
+            starting with `#`. If array-like must be 1-dimensional array with 3
+            or 4 elements. If a list is supplied it must be the same length as
+            the length of `data` and each element will be applied to each shape
+            otherwise the same value will be used for all shapes.
+        face_color : str | tuple | list
+            If string can be any color name recognized by vispy or hex value if
+            starting with `#`. If array-like must be 1-dimensional array with 3
+            or 4 elements. If a list is supplied it must be the same length as
+            the length of `data` and each element will be applied to each shape
+            otherwise the same value will be used for all shapes.
+        opacity : float | list
+            Opacity of the shapes, must be between 0 and 1.
+        z_index : int | list
+            Specifier of z order priority. Shapes with higher z order are
+            displayed ontop of others. If a list is supplied it must be the
+            same length as the length of `data` and each element will be
+            applied to each shape otherwise the same value will be used for all
+            shapes.
+        """
+        if len(data) == 0:
+            return
+
+        if np.array(data[0]).ndim == 1:
+            # If a single array for a shape has been passed
+            if shape_type in self.data._types.keys():
+                shape_cls = self.data._types[shape_type]
+                shape = shape_cls(data, edge_width=edge_width,
+                                  edge_color=edge_color, face_color=face_color,
+                                  opacity=opacity, z_index=z_index)
+            else:
+                raise ValueError("""shape_type not recognized, must be one of
+                                 "{'line', 'rectangle', 'ellipse', 'path',
+                                 'polygon'}"
+                                 """)
+            self.data.add(shape)
+        else:
+            # Turn input arguments into iterables
+            shape_types = ensure_iterable(shape_type)
+            edge_widths = ensure_iterable(edge_width)
+            opacities = ensure_iterable(opacity)
+            z_indices = ensure_iterable(z_index)
+            edge_colors = ensure_iterable(edge_color, color=True)
+            face_colors = ensure_iterable(face_color, color=True)
+
+            for d, st, ew, ec, fc, o, z, in zip(data, shape_types, edge_widths,
```

Comment:
I'm still hesitant about this. What id someone misses one edge color in the list, this entire init will fail.

Reply:
It won't. `zip` stops at the shortest list, so if you miss an edge color, you'll miss a shape. I think that is the correct behaviour, +/- a warning.